File: 백준/Silver/21920. 서로소 평균/서로소 평균.swift
import Foundation

let N = Int(readLine()!)!
let arr = readLine()!.split(separator: " ").map{Int($0)!}
let X = Int(readLine()!)!
var total = 0
var cnt = 0

func euc (A:Int, B:Int) -> Int{
    let remain = A%B
    if remain == 0{
        return B
    }
    return euc(A: B, B: remain)
}

for num in arr{
    if euc(A: num, B: X)==1{
        total += num
        cnt += 1
    }
}
let ans = Double(total)/Double(cnt)
print(ans)

File: 백준/Silver/21920. 서로소 평균/서로소 평균.cc
#include <iostream>
#include <vector>
using namespace std;

int euc(int A, int B){
    int remain = A%B;
    if (remain == 0){
        return B;
    }
    return euc(B, remain);
}

int main() {
    vector<int> arr;
    int N, X;
    long long total = 0;
    int cnt = 0;
    double ans;
    
    cin >> N;
    for (int i=0; i<N; i++){
        int num;
        cin >> num;
        arr.push_back(num);
    }
    cin >> X;
    
    for(auto num: arr){
        if (euc(X, num)==1){
            cnt += 1;
            total += num;
        }
    }
    ans = double(total)/double(cnt);
    cout << ans << endl;
    
    return 0;
}